### PR TITLE
feat(auth): add stub for BasePasswordHasher decode method and salt_entropy variable

### PR DESCRIPTION
**Pull request by Kadermiyanyedi**
Hi! I was looking at the Todo list and thought I could help with this.

* I made additions for the salt_entropy variable and the decode method in the BasePasswordHasher class. 
*Also removed the algorithm and variable variables that were added before but included in the todo list.

### CODE REVIEW

**Review by sobolevn — COMMENTED** [review]
Hi, thanks a lot for your help!

But, there are some problems yet to fix:

```
error: django.contrib.auth.hashers.BasePasswordHasher.algorithm variable differs from runtime type None
Stub: in file /home/runner/work/django-stubs/django-stubs/django-stubs/contrib/auth/hashers.pyi:18
builtins.str
Runtime:
None

error: django.contrib.auth.hashers.BasePasswordHasher.library variable differs from runtime type None
Stub: in file /home/runner/work/django-stubs/django-stubs/django-stubs/contrib/auth/hashers.pyi:19
Union[builtins.str, tuple[builtins.str, builtins.str]]
Runtime:
None

note: unused allowlist entry django.contrib.auth.hashers.Argon2PasswordHasher.decode
note: unused allowlist entry django.contrib.auth.hashers.BCryptSHA256PasswordHasher.decode
note: unused allowlist entry django.contrib.auth.hashers.CryptPasswordHasher.decode
note: unused allowlist entry django.contrib.auth.hashers.MD5PasswordHasher.decode
note: unused allowlist entry django.contrib.auth.hashers.PBKDF2PasswordHasher.decode
note: unused allowlist entry django.contrib.auth.hashers.SHA1PasswordHasher.decode
note: unused allowlist entry django.contrib.auth.hashers.UnsaltedMD5PasswordHasher.decode
note: unused allowlist entry django.contrib.auth.hashers.UnsaltedSHA1PasswordHasher.decode
```

**Comment by Kadermiyanyedi** [comment]
Hi, thanks for your response. I checked the Django source code again.

-> I added the decode function to the subclasses if the function return type is different from the BasePasswordHasher class

-> The algorithm and library variables are set to None in the BasePasswordHasher class in Django source code. The library variable is set to string or tuple[string] only in BCryptSHA256PasswordHasher and Argon2PasswordHasher classes.  I set to None the library variable in the BasePasswordHasher class to None and I added library variable to two sub classes.

And finally I removed these changes from the todo list file.

File: django-stubs/contrib/auth/hashers.pyi [hunk 1]
```diff
@@ -15,30 +15,55 @@ def identify_hasher(encoded: str) -> BasePasswordHasher: ...
 def mask_hash(hash: str, show: int = ..., char: str = ...) -> str: ...
 
 class BasePasswordHasher:
-    algorithm: str
-    library: str | tuple[str, str]
+    algorithm: None
+    library: None
```
Comment (sobolevn) [inline django-stubs/contrib/auth/hashers.pyi]:
```suggestion
    library: library: str | tuple[str, str] | None
```

File: django-stubs/contrib/auth/hashers.pyi [hunk 2]
```diff
@@ -15,30 +15,55 @@ def identify_hasher(encoded: str) -> BasePasswordHasher: ...
 def mask_hash(hash: str, show: int = ..., char: str = ...) -> str: ...
 
 class BasePasswordHasher:
-    algorithm: str
-    library: str | tuple[str, str]
+    algorithm: None
```
Comment (sobolevn) [inline django-stubs/contrib/auth/hashers.pyi]:
```suggestion
    algorithm: str | None
```

**Review by sobolevn — COMMENTED** [review]
Maybe like this?

File: django-stubs/contrib/auth/hashers.pyi (hunk 1 above)
Comment (Kadermiyanyedi) [reply django-stubs/contrib/auth/hashers.pyi]: I applied this changes, thanks for your suggestion

File: django-stubs/contrib/auth/hashers.pyi (hunk 2 above)
Comment (Kadermiyanyedi) [reply django-stubs/contrib/auth/hashers.pyi]: I applied this change, thanks for your suggestion :)

**Comment by Kadermiyanyedi** [comment]
Hi, @sobolevn do you have any other suggestions? I can do :)

**Review by sobolevn — APPROVED** [review]
No, LGTM! Thank you

**Pull request merged**